assets/src/edit-story/components/panels/textStyle/font.js

```diff
@@ -47,16 +47,53 @@ function FontControls({ selectedElements, pushUpdate }) {
   const fontFamily = getCommonValue(selectedElements, 'fontFamily');
   const fontSize = getCommonValue(selectedElements, 'fontSize');
   const fontWeight = getCommonValue(selectedElements, 'fontWeight');
+  const timeout = useRef(null);
 
   const {
     state: { fonts },
-    actions: { getFontWeight, getFontFallback },
+    actions: { maybeEnqueueFontStyle, getFontWeight, getFontFallback },
   } = useFont();
   const fontWeights = useMemo(() => getFontWeight(fontFamily), [
     getFontWeight,
     fontFamily,
   ]);
 
+  const loadFont = useCallback(
```

Comment:
This is too much code for this module. This should be a utility. Not sure if it needs to be combined with `maybeEnqueueFontStyle`, but at least all the `document.check/load` stuff.

Reply:
Agreed, as @barklund already suggested this, that's would be interesting to make reusable.